### ISSUE TITLE
How do i get the size of preds and truth tensors after i call .update()

### ISSUE THREAD

**Issue by P-Sood** [Lightning-AI/metrics#1326]
### Discussed in https://github.com/Lightning-AI/metrics/discussions/1325

<div type='discussions-op-text'>

<sup>Originally posted by **P-Sood** November  9, 2022</sup>
I am running my model using DDP, and 4 GPUS, and I want to make sure that the metrics being computed are consistent by checking their size after and before a reset, as well as seeing how they line up with my batch size</div>

**Comment by SkafteNicki** [comment]
Hi @P-Sood,
Is there a specific metric you are thinking about here or are we talking about a custom metric of yours?

**Comment by P-Sood** [comment]
This would be on any of the Multi class ones for Accuracy , F1 , Precison , Recall or Confusion matrix.

I am scared that my GPU's arent talking to each other and I would like a way to make sure before I call the compute function and after, that my code is correct and I don't have a silly bug in there.

**Comment by SkafteNicki** [comment]
So by default we do not track those kind of statistics, but with a bit of custom code it should be possible to get the stats. In the example below I am simply adding a statistic that keeps track of how many samples the metric has seen (over all devices)
```python
from torchmetrics.classification import MulticlassAccuracy
import torch

class MyMulticlassAccuracy(MulticlassAccuracy):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.add_state("number_of_samples", torch.zeros(1), dist_reduce_fx="sum")

    def update(self, preds, target):
        super().update(preds, target)
        self.number_of_samples += preds.shape[0]


if __name__ == "__main__":
    acc = MyMulticlassAccuracy(num_classes=5)
    # 10 samples per batch
    preds = torch.randn(10, 5)
    target = torch.randint(5, (10,))

    # 10 batches
    for _ in range(10):
        acc.update(preds, target)
    
    val = acc.compute()
    assert acc.number_of_samples == 100
```
Is it something like this you are looking for?
